lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -287,6 +310,36 @@ fn parse_dogstatsd<'a>(
     ))
 }
 
+fn parse_dogstatsd_event<'a>(input: &'a [u8], _config: &DogstatsdCodecConfiguration) -> IResult<&'a [u8], EventD> {
+    let event_header_parser = map(take_while1(|c| c != b':'), |bytes: &[u8]| unsafe {
+        std::str::from_utf8_unchecked(bytes)
+    });
+
+    let event_data_parser = map(
+        take_while1(|c: u8| c.is_ascii_alphanumeric() || c == b' ' || c == b'|' || c == b'_' || c == b'-' || c == b'.'),
+        |bytes: &[u8]| unsafe { std::str::from_utf8_unchecked(bytes) },
+    );
```

Comment:
I think you could boil this down heavily and get rid of `parse_dogstatsd_event_header` entirely. `nom` makes it trivial to handle this sort of thing, and you've already got some of the underpinning below.

Essentially, what do we want out of the header? The title and text length. Their structure is well-formed: a prefix, a length, a separator, a length, and a suffix... so we just match that directly:

```rust
// In the imports:
use nom::character::complete::u32 as parse_u32;

// Down in `parse_dogstatsd_event`:
let (remaining, (title_len, text_len)) = terminated(delimited(EVENT_PREFIX, separated_pair(parse_u32, tag(b":"), parse_u32), tag(b"}:"))?;
```

After that, you've got `<title>|<text>|......` left, which you can use `take` for:

```rust
// In the imports:
use nom::bytes::complete::take;

// Down in `parse_dogstatsd_event`:
let (remaining, (raw_title, raw_text)) = separated_pair(take(title_len), tag(b"|"), take(text_len))?
```

Now you've got the raw byte slices for title and text, which you can try converting to UTF-8: look at our usage of `simdutf8::basic::from_utf8` in `metric_value` for how to do that.

After that, you've just got your list of optional values, which you can parse by reusing the pattern of how we parse optional fields in the metrics payloads.